src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -85,16 +89,17 @@ internal Solution(Workspace workspace, SolutionInfo.SolutionAttributes solutionA
         /// <summary>
         /// True if the solution contains a project with the specified project ID.
         /// </summary>
-        public bool ContainsProject(ProjectId projectId) => _state.ContainsProject(projectId);
+        public bool ContainsProject([NotNullWhen(returnValue: true)] ProjectId? projectId) => _state.ContainsProject(projectId);
 
         /// <summary>
         /// Gets the project in this solution with the specified project ID. 
         /// 
         /// If the id is not an id of a project that is part of this solution the method returns null.
         /// </summary>
-        public Project GetProject(ProjectId projectId)
+        // TODO: is it possible to annotate this method to say that if it returned non-null, then
+        // the input was non-null?
+        public Project? GetProject(ProjectId? projectId)
         {
-            // ContainsProject checks projectId being null
```

Comment:
Comment is removed now that ContainsProject's signature carries this fact directly.